src/Repository/WorklogRepository.php

```diff
@@ -113,6 +113,79 @@ public function findWorklogsByWorkerAndDateRange(string $workerIdentifier, \Date
             ->getQuery()->getResult();
     }
 
+    /**
+     * Gets the sum of time spent grouped by week of year for a specific worker within a range of weeks.
+     *
+     * @param string $workerEmail The email of the worker
+     * @param int $fromWeek The starting week number
+     * @param int $toWeek The ending week number
+     * @param string $groupBy The function to group by, accepts 'week', 'month' and 'year'
+     *
+     * @return array An array of results containing total time spent, week number, and worker
+     */
+    public function getTimeSpentByWorkerInWeekRange(
+        string $workerEmail,
+        \DateTimeInterface $from,
+        \DateTimeInterface $to,
+        string $groupBy,
+    ): array {
+        $qb = $this->createQueryBuilder('w');
+
+        if ('month' !== $groupBy && 'week' !== $groupBy && 'year' !== $groupBy) {
+            throw new \InvalidArgumentException('Invalid group by parameter, function accepts only "week" or "month"');
+        }
+
+        $groupByFunction = 'week' === $groupBy ? 'WEEKOFYEAR' : \strtoupper($groupBy);
+        $dqlPart = sprintf('%s(w.started) as %s', $groupByFunction, $groupBy);
+
+        $qb
+            ->select(
+                'SUM(w.timeSpentSeconds) as totalTimeSpent',
+                $dqlPart,
+                'w.worker'
+            )
+            ->where('w.worker = :worker')
+            ->andWhere('w.started >= :from')
+            ->andWhere('w.started <= :to')
+            ->setParameter('worker', $workerEmail)
+            ->setParameter('from', $from)
+            ->setParameter('to', $to)
+            ->groupBy($groupBy)
+            ->orderBy('w.started', 'ASC');
+
+        return $qb->getQuery()->getResult();
+    }
+
+    /**
+     * Gets the sum of time spent grouped by week of year for a specific worker within a range of weeks.
+     *
+     * @param string $workerEmail The email of the worker
+     * @param int $fromWeek The starting week number
+     * @param int $toWeek The ending week number
+     * @param string $groupBy The function to group by, accepts 'week', 'month' and 'year'
+     *
+     * @return array An array of results containing total time spent, week number, and worker
+     */
+    public function testDateTime(
```

Comment:
Is this use for anything?

Reply:
Removed. It was for validating returned results from db with `WHERE => [datetime]`